ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Types.hs

```diff
@@ -800,36 +801,47 @@ connectionManagerErrorFromException x = do
-  | TrUnregisterConnection       !Provenance !peerAddr
-  | TrConnect                    !(Maybe peerAddr) -- ^ local address
-                                 !peerAddr         -- ^ remote address
-  | TrConnectError               !(Maybe peerAddr) -- ^ local address
-                                 !peerAddr         -- ^ remote address
-                                 !SomeException
-  | TrTerminatingConnection      !Provenance !(ConnectionId peerAddr)
-  | TrTerminatedConnection       !Provenance !peerAddr
-  | TrConnectionHandler          !(ConnectionId peerAddr) !handlerTrace
+  = TrIncludeConnection            !Provenance !peerAddr
+  | TrUnregisterConnection         !Provenance !peerAddr
+  | TrConnect                      !(Maybe peerAddr) -- ^ local address
+                                   !peerAddr         -- ^ remote address
+  | TrConnectError                 !(Maybe peerAddr) -- ^ local address
+                                   !peerAddr         -- ^ remote address
+                                   !SomeException
+  | TrTerminatingConnection        !Provenance !(ConnectionId peerAddr)
+  | TrTerminatedConnection         !Provenance !peerAddr
+  | TrConnectionHandler            !(ConnectionId peerAddr) !handlerTrace
   | TrShutdown
-  | TrConnectionExists           !Provenance !peerAddr    !AbstractState
-  | TrForbiddenConnection        !(ConnectionId peerAddr)
-  | TrImpossibleConnection       !(ConnectionId peerAddr)
-  | TrConnectionFailure          !(ConnectionId peerAddr)
-  | TrConnectionNotFound         !Provenance !peerAddr
-  | TrForbiddenOperation         !peerAddr                !AbstractState
-  | TrPruneConnections           ![peerAddr]
-  | TrConnectionCleanup          !(ConnectionId peerAddr)
-  | TrConnectionTimeWait         !(ConnectionId peerAddr)
-  | TrConnectionTimeWaitDone     !(ConnectionId peerAddr)
-  | TrConnectionManagerCounters  !ConnectionManagerCounters
-  | TrState                      !(Map peerAddr AbstractState)
+  | TrConnectionExists             !Provenance !peerAddr    !AbstractState
+  | TrForbiddenConnection          !(ConnectionId peerAddr)
+  | TrImpossibleConnection         !(ConnectionId peerAddr)
+  | TrConnectionFailure            !(ConnectionId peerAddr)
+  | TrConnectionNotFound           !Provenance !peerAddr
+  | TrForbiddenOperation           !peerAddr                !AbstractState
+  | TrPruneConnections             ![peerAddr]
+  | TrConnectionCleanup            !(ConnectionId peerAddr)
+  | TrConnectionTimeWait           !(ConnectionId peerAddr)
+  | TrConnectionTimeWaitDone       !(ConnectionId peerAddr)
+  | TrConnectionManagerCounters    !ConnectionManagerCounters
+  | TrState                        !(Map peerAddr AbstractState)
   -- ^ traced on SIGUSR1 signal, installed in 'runDataDiffusion'
-  | TrUnexpectedlyMissingConnectionState !(ConnectionId peerAddr)
+  | TrCMUnexpectedlyFalseAssertion !(AssertionLocation peerAddr)
```

Comment:
```suggestion
  | TrUnexpectedlyFalseAssertion !(AssertionLocation peerAddr)
```
I plan for the types in `connection-manager`, `inbound-governor` etc. to be always imported qualified.

Reply:
I see, I left the name CM/IG there so one could check in which component was this traced. Is there any way to reliably know without the names?

Reply:
yes, the `cardano-node` tracer will include tracer name in its context.

Reply:
I think it is derived from the type, so we might need to check how `ouroboros-consensus` is dealing with this (many tracers there are just called `Tracer`).